File: libs/redis/src/schemas/customvar.ts
import { CustomVarType } from '@tsuwari/prisma';
import { Entity, Schema } from 'redis-om';

export class CustomVar extends Entity {
  id: string;
  name: string;
  description: string | null;
  type: CustomVarType;
  evalValue: string | null;
  response: string | null;
  channelId: string;
}

export const customVarSchema = new Schema(CustomVar, {
  id: { type: 'string' },
  name: { type: 'string' },
  description: { type: 'string' },
  type: { type: 'string' },
  evalValue: { type: 'string' },
  response: { type: 'string' },
  channelId: { type: 'string' },
}, {
  prefix: 'variables',
  indexedDefault: true,
});

File: apps/api-gql/internal/entity/customvar.go
package entity

import (
	"github.com/google/uuid"
)

type CustomVarType string

const (
	CustomVarScript CustomVarType = "SCRIPT"
	CustomVarText   CustomVarType = "TEXT"
	CustomVarNumber CustomVarType = "NUMBER"
)

type CustomVarScriptLanguage string

func (s CustomVarScriptLanguage) String() string {
	return string(s)
}

const (
	ScriptLanguageJavaScript = "javascript"
	ScriptLanguagePython     = "python"
)

type CustomVariable struct {
	ID             uuid.UUID
	ChannelID      string
	Name           string
	Description    *string
	Type           CustomVarType
	EvalValue      string
	Response       string
	ScriptLanguage CustomVarScriptLanguage
}

var CustomVarNil = CustomVariable{}
